### PR TITLE
ipn/localapi: also allow localhost as the LocalAPI host

### PR DESCRIPTION
**Pull request by mihaip**
The Mac and iOS LocalAPI clients make requests to it.

Slightly relaxes the checks added by #6346.

### CODE REVIEW

File: ipn/localapi/localapi.go [hunk 1]
```diff
@@ -168,12 +168,23 @@ func validHost(h string) bool {
 	case "", apitype.LocalAPIHost:
```
Comment (maisem) [inline ipn/localapi/localapi.go:168]: can you do it here?

Comment (maisem) [reply ipn/localapi/localapi.go:168]: or do they also specify a port?

Comment (mihaip) [reply ipn/localapi/localapi.go:168]: They specify a port: https://github.com/tailscale/tailscale/blob/1ea43d4c87931c74af28209ff1689cd53d291ca1/client/tailscale/localclient.go#L94-L100 and https://github.com/tailscale/corp/blob/c50d5ead9d002d652c0db56449bd720d336c5ed9/xcode/Shared/LocalAPIClient.swift#L307

File: ipn/localapi/localapi.go [hunk 2]
```diff
@@ -168,12 +168,23 @@ func validHost(h string) bool {
...
+	if err != nil {
+		return false
+	}
+	if runtime.GOOS == "windows" && port != safesocket.WindowsLocalPort {
+		return false
+	}
+	if host == "localhost" {
+		return true
+	}
+	ap, err := netip.ParseAddrPort(h)
```
Comment (maisem) [inline ipn/localapi/localapi.go:186]: you could make this just be ParseAddr now.

Comment (mihaip) [reply ipn/localapi/localapi.go]: Done.

File: ipn/localapi/localapi.go [hunk 3]
```diff
@@ -168,15 +168,26 @@ func validHost(h string) bool {
...
 	}
-	// Otherwise, any Host header we see should at most be an ip:port.
-	ap, err := netip.ParseAddrPort(h)
+	// Allow either localhost or loopback IP hosts.
+	host, portStr, err := net.SplitHostPort(h)
 	if err != nil {
 		return false
 	}
-	if runtime.GOOS == "windows" && ap.Port() != safesocket.WindowsLocalPort {
+	port, err := strconv.Atoi(portStr)
```
Comment (bradfitz) [inline ipn/localapi/localapi.go:176]: technically a uint16, not an int.

strconv.ParseUint(...)

Comment (mihaip) [reply ipn/localapi/localapi.go]: Done.

**Review by maisem — APPROVED** [review]

**Review by danderson — APPROVED** [review]

**Pull request merged**